File: content/examples/other/leetcode/problems/jump-game/example-1/solution.js
function canJumpDP(nums) {
  const good = new Array(nums.length).fill(false);
  good[nums.length - 1] = true;
  for (let i = nums.length - 2; i >= 0; i -= 1) {
    const farthest = Math.min(nums.length - 1, i + nums[i]);
    for (let j = i + 1; j <= farthest; j += 1) {
      if (good[j]) {
        good[i] = true;
        break;
      }
    }
  }
  return good[0];
}

if (require.main === module) {
  console.log(canJumpDP([2, 3, 1, 1, 4]));
  console.log(canJumpDP([3, 2, 1, 0, 4]));
}

module.exports = { canJumpDP };

File: content/examples/other/leetcode/problems/jump-game/example-1/Solution.java
public class Solution {
  static boolean canJumpDP(int[] nums) {
    boolean[] good = new boolean[nums.length];
    good[nums.length - 1] = true;
    for (int i = nums.length - 2; i >= 0; i -= 1) {
      int farthest = Math.min(nums.length - 1, i + nums[i]);
      for (int j = i + 1; j <= farthest; j += 1) {
        if (good[j]) {
          good[i] = true;
          break;
        }
      }
    }
    return good[0];
  }

  public static void main(String[] args) {
    System.out.println(canJumpDP(new int[] {2, 3, 1, 1, 4}));
    System.out.println(canJumpDP(new int[] {3, 2, 1, 0, 4}));
  }
}
